app/jobs/create_uphold_cards_job.rb

```diff
@@ -9,16 +9,37 @@ def perform(uphold_connection_id:)
       return
     end
 
-    card = nil
+    # We create the cards, so let's first check to see if the publisher has an address so we might not have to change anything
+    if uphold_connection.address.present?
+      begin
+        card = uphold_connection.uphold_client.card.find(uphold_connection: uphold_connection, id: uphold_connection.address)
+
+        return if card&.currency.eql?(uphold_connection.default_currency)
+      rescue Faraday::ResourceNotFound
+        # This most likely won't ever happen because it's not possible to delete a card. Better safe than sorry.
+      end
+    end
 
     # Search for an existing card
     cards = uphold_connection.uphold_client.card.where(uphold_connection: uphold_connection)
     existing_private_cards = UpholdConnectionForChannel.select(:card_id).where(uphold_connection: uphold_connection).to_a
 
+    # This is the default label that we apply to a card. If we find it then it's safe to assume that this was one we've created prior.
+    card = cards.detect { |c| c.label.eql?("Brave Rewards") }
+
+    # User's can change the label's on their cards so if we couldn't find it, we'll have to iterate until we find a card that matches are criteria
+    # 1. Isn't the browser's wallet card
+    # 2. Isn't a channel card
```

Comment:
Might be better to extract this block to its own function. Also only call `if card.present?`

Reply:
Fixed things up - I keep in the `break if card.present?` because keeping it as is was a little bit simpler than rewriting the loop